index.js

```diff
@@ -42,21 +76,42 @@ var
     return cWhere (collection, predicates);
   };
 
-var objectList = function objectList (collection) {
-  return {
-    getByKey: function (key) {
-      return getByKey.apply(null, [collection, key]);
-    },
-    whitelist: function (keyWhitelist) {
-      return whitelist.apply(null, [collection, keyWhitelist]);
-    },
-    concat: function () {
-      return concat(collection);
-    },
-    where: function (keyWhitelist) {
-      return where.apply(null, [collection, keyWhitelist]);
-    },
-  };
+objectList = function objectList (options) {
+  var
+    collection,
+    api = {
+      getByKey: function (key) {
+        return getByKey.apply(null, [collection, key]);
+      },
+      whitelist: function (keyWhitelist) {
+        return whitelist.apply(null, [collection, keyWhitelist]);
+      },
+      concat: function () {
+        return concat(collection);
+      },
+      where: function (keyWhitelist) {
+        return where.apply(null, [collection, keyWhitelist]);
+      },
+      add: function (records) {
+        return add.apply(null, [collection, records]);
+      },
+      push: function () {
+        return api.add.apply(null, [].slice.call(arguments));
```

Comment:
I didn't because I thought I remembered that `arguments` doesn't work with `.apply()` since it's not a true array, but I'm probably misremembering. I'll try it without the slice.